dspace-api/src/main/java/org/dspace/authorize/AuthorizeServiceImpl.java

```diff
@@ -697,7 +697,7 @@ public ResourcePolicy createOrModifyPolicy(ResourcePolicy policy, Context contex
             if (!duplicates.isEmpty()) {
                 policy = duplicates.get(0);
             }
-        } else {
+        } else if (group != null) {
```

Comment:
This was a situation where Hibernate 6 is much less "forgiving" of null values in queries.  This block of code runs `findByTypeGroupAction()`, but if the `group` is null this throws errors in Hibernate.  That's why this check had to be modified to first verify that the `group` is not null (as otherwise tests which only passed in an EPerson would fail with Hibernate errors).

So, I think this change is correct and isn't an indication of a bug. The code in the "else if" seems to suggest that a `group` is required based on this comment:
> // if an identical policy (same Action and same Group) is already in place modify it

Reply:
Would've preferred Hibernate to call it an illegal-argument error but I like the new behavior better.  And we should not even ask Hibernate if we are asking for nothing.